### ISSUE TITLE
suggest remote backup storage options?

### ISSUE THREAD

**Issue by ThomasWaldmann** [borgbackup/borg#2177]
- http://www.rsync.net/products/attic.html (ssh account with borg, managed, storage on zfs raidz3)
- https://www.time4vps.eu/pricing/ "High Volume" tab (self-managed vps, storage on 12 disk raid6)
- https://www.kimsufi.com/de/server.xml e.g. KS-2E (self-managed root server, single disk)
- amazon clouddrive together with http://rclone.org/ (needs local borg backup first)

**Comment by ThomasWaldmann** [comment]
After we collected some options, this ticket could be closed by moving this to a section on the community repo link list.

**Comment by twink0r** [comment]
All VPS based on KVM
https://spartanhost.org/vps#storage
https://backupsy.com/
https://buyvm.net/storage-vps/
https://hosthatch.com/storage-vps

**Comment by ThomasWaldmann** [comment]
@twink0r hmm, some of these don't seem that interesting (looking at price) when comparing to e.g. rsync.net. add some details?

**Comment by enkore** [comment]
https://www.hetzner.de/hosting/produktmatrix/storagebox-produktmatrix

**Comment by milkey-mouse** [comment]
[rclone now has a mount option](http://rclone.org/commands/rclone_mount/) that mounts the backend as a FUSE filesystem. Theoretically one could just mount and backup directly into it, though at least for Amazon Drive, Amazon's "eventual consistency" stuff causes problems like files spontaneously disappearing after upload, only to reappear 30 seconds later (which is acceptable for an online cloud storage website, but not so much for a filesystem).

I was also thinking about the fact that many cloud providers theoretically offer "unlimited" storage, but throttle & have a limit in the fine print. According to [an /r/DataHoarder thread](https://www.reddit.com/r/DataHoarder/comments/3xgsdc/is_amazon_cloud_drive_safe/cy5g6u7/) it seems relatively safe to store one's backups on Amazon Drive, provided you encrypt first.

And, of course, there are always small VPS instances from Azure, Google Cloud, AWS, Vultr, etc. which can just run `borg serve` on an SSH connection.

**Comment by Vertux** [comment]
For me the most appealing offer regarding value/price is Amazon S3 Standard or Glacier for long term backups. And another plus is, that you can choose the location where your data is stored e.g. Frankfurt.

**Comment by DArtagan** [comment]
I think these people have done an interesting thing: borg does the backups and then they use [git-annex](https://git-annex.branchable.com/) to keep an index of the chunks and manage syncing the chunks to remote storage (e.g. B2).

I think the first one has an especially good description, the second is more skimmable:
* http://zyradyl.github.io/2016/10/02/Collegi-Backup-System/
* https://juliank.wordpress.com/2016/05/11/backing-up-with-borg-and-git-annex/

If we can move away from VPS-type options, that eliminates a common cost.  From what I can tell, using this combination opens up inexpensive storage options like B2.  How do you think this approach stacks up against rclone?

**Comment by willsALMANJ** [comment]
It seems like #1795 can be closed in favor of this.

The posts about using borg with git-annex, rclone, etc. are good to know about, but not particularly special. They just describing normal borg use -- backing up to a local disk -- and then using another tool to copy files to cloud storage.

I think rsync.net is definitely worth highlighting. It is awesome that they provide such a customized borg offering. It would be nice to highlight some VPS's that have reasonably priced large storage options, though I don't know how often such a list would need to be curated. I don't see much value in highlighting making a local backup and copying it to the cloud since that can be done with pretty much any service. Maybe one day #102 will be closed and more cloud storage options will work better with borg.

**Comment by DArtagan** [comment]
The day #102 closes will be a good day indeed.  

Wow, #1795 has the same git-annex links as I included, and here I thought I'd uncovered two previously unknown gems.  

However, I do think making a local backup and copying it to the cloud is hugely important: if done right it breaks us from needing a VPS (higher cost/complexity).  But doing it right is the catch, just copying the files doesn't prevent something going wrong with the source repo that you're copying from.  That's why the [borg documentation](http://borgbackup.readthedocs.io/en/stable/faq.html#can-i-copy-or-synchronize-my-repo-to-another-location) talks about having redundant, independent repositories.  This is where git-annex comes in, as an alternative to the VPS/rsync.net/SSH approach.  As crazy as this sounds: it's version control for your backups.  Keeping the backups version controlled allows the use of dumb, cloud storage.  At least... that's what I'm thinking, but I welcome further consideration.

**Comment by markfox1** [comment]
@DArtagan is thinking in the right direction here. Requiring a VPS adds expense and complication.

I'm on the tail end of a four-day deep dive into backup systems. (There were earlier dives, but no so deep.) The best solutions I have been able to find are [HashBackup](https://www.hashbackup.com), [Tarsnap](https://www.tarsnap.com), and [Duplicati](https://duplicati.com).

All are client-side only, in the sense that there is no need to read data back from the remote site when doing a backup. (This is actually not completely true, but very close.) In a nutshell, they do local book-keeping so that they can dedup, encrypt, and send only incremental changes to the remote store. So the first backup is the only painful one.

Each has its flaw too. Neither HashBackup or Tarsnap are open source. HashBackup, although free for the moment, may adopt a license in the $250/year range. Tarsnap is actually quite expensive at $0.25/GB/month stored and $0.25/GB of bandwidth. Duplicati goofed on fixed block offsets vs checksum-based offsets and the security model could be better. (Keep in mind the implications of a compromise of the backed up box.)

But the algorithms each use can make simple backups no more expensive than the cloud-storage and the bandwidth required to get them there. We're talking around $5US/TB/month. That price opens up a lot of doors. It's a big deal.

Borg is well positioned here. It could be _the_ backup tool in fairly short order.

**Comment by ThomasWaldmann** [comment]
@enkore do you use hetzner storagebox yourself? which protocol / filesystem?

**Comment by enkore** [comment]
I haven't rented servers in a while. Someone told me they use it with iirc sshfs or sambda.

**Comment by wzyboy** [comment]
I am a new user (~4 days of usage) of BorgBackup. After reading this thread I started to using B2 (BackBlade's S3) as an offsite backup. I have to admit I really enjoying using B2 -- much lower price than S3/Glacier, reasonable daily free tier usage quota, and rather satisfying uploading speed.

Here's my setup (complies with BackBlade's [3-2-1 backup strategy](https://www.backblaze.com/blog/the-3-2-1-backup-strategy/)):

- I have a HP Gen8 microserver with 128 SSD and 2 TB + 4 TB HDD. (JBOD, with [mergerfs](https://github.com/trapexit/mergerfs) for painless "unionfs-like" access);
- A systemd timer on my laptop runs borgmatic daily, backing up `/home` and `/etc` to the home server via SSH;
- From the home server, `rclone` uploads the Borg repo to B2. Because Borg never "modifies" a data file (it just "creates" or "deletes" one), it's rather safe even if a `borg create` is in progress when a `rclone sync` job is running.

Note that [the current version of rclone uses a new sync method that exhausts your daily free API calls quota quickly if you have a lot of directories](https://forum.rclone.org/t/new-sync-method-in-v1-36-and-backblaze-b2-class-c-transaction-cap/1441). With Borg repo, however, this does not pose an issue, since a Borg repo has very few directories.

**Comment by willsALMANJ** [comment]
I will experiment with `rclone sync` and `rclone mount`. My reservations regarding `rclone sync` and `git annex` style options is that for local backup I prefer a plain copy of the files with some kind of snapshot system because that is the easiest to access, so making a local borg backup and then copying it to the cloud means I need to have two local backups if I keep my file system backup, though maybe having an extra backup is a good thing.

**Comment by mannp** [comment]
S3 Compatible would open up a lot of alternatives, including using minio as a backend.

**Comment by mannp** [comment]
@wzyboy Have you tested restores and how that would work in your case?

I have the same setup and use Backblaze, but not with borg (yet) and I wondered what happens, say if your local repository becomes corrupted, can rclone just restore the broken parts from your local copy?

I am trying to minimise the download costs from B2 if I have a failure of my local repo.

**Comment by wzyboy** [comment]
@mannp Rclone only copies necessary data whenever possible. Say if 10% of your local backup is corrupted, only 10% data is needed to be downloaded from B2 to repair your local backup.

In desperate situations, you can even `rclone mount` your B2 bucket and `borg mount` on top of that. The performance of stacked FUSE might be terrible, though.

**Comment by mannp** [comment]
@wzyboy ok thanks for that; yes I assume it would only copy what was necessary, but wondered if those files were copied back, than borg would be able to recover. I assume it would if the files were restored the same. Not tried though..

Do you try and restrict the B2 transactions, so they remain within your daily allowance, with --fast-list or the like options?

Do you mind sharing your rclone command line?

Good point, I will try the mount, but was always concerned about how many transactions a mount would incur... have you tried it?

Thanks

**Comment by wzyboy** [comment]
@mannp 

> ok thanks for that; yes I assume it would only copy what was necessary, but wondered if those files were copied back, than borg would be able to recover. I assume it would if the files were restored the same. Not tried though..

rclone works bi-directionally.

> Do you try and restrict the B2 transactions, so they remain within your daily allowance, with --fast-list or the like options?

No. I have never exceeded my daily transaction quota.

> Do you mind sharing your rclone command line?

```
rclone -v sync ${REPO} b2:${BUCKET}
```

> Good point, I will try the mount, but was always concerned about how many transactions a mount would incur... have you tried it?

I suppose in a desperate situation you would not care about how much transaction fee you would pay BackBlaze :-)

**Comment by mannp** [comment]
@wzyboy Ok cool, thanks for taking the time to respond 👍 

No command line extras at all, ok fair enough, thanks again.

**Comment by mannp** [comment]
@wzyboy Can't find any info on rclone supporting two way sync? So you'd have to download the whole repo, if the local one was lost....

**Comment by Orbixx** [comment]
I've been using https://i-83.net/ which provides NAT'd IPv4 and native IPv6, very cheap storage VPS. Occasionally goes down for what is presumably maintenance, but works well for a daily Borg backup.

Here's a couple of providers I'm also considering but have not tried - be curious to know if anybody has used them:

https://www.wishosting.com/order/main/packages/VPS/?group_id=2 (see the OpenVZ Mini package)

https://backupsy.com/ (see the 40% off for life coupon, also 250GB minimum. 250GB storage for $6/mo)

**Comment by ThomasWaldmann** [comment]
General note: be careful with memory amount, the repo index needs to fit into RAM. So if your repo has a lot of chunks, it might not fit. See the docs for a formula.

**Comment by trendco** [comment]
At the moment i try the 1TB HiDrive from Strato: https://www.strato.de/online-speicher/

**Comment by ThomasWaldmann** [comment]
note: trendco also noted on irc, that he uses the hidrive via sshfs and that strato creates user-accessible backups of the hidrive. will be interesting to hear practical experiences with that, like speed, reliability, etc.

**Comment by Cardes** [comment]
Hetzners Storageboxes now offer Borg Backup as a service / repository:

[Hetzners Wiki Entry](https://wiki.hetzner.de/index.php/BorgBackup/en)

**Comment by ThomasWaldmann** [comment]
https://contabo.de/?show=vps (from @bebehei on irc)

**Comment by bebehei** [comment]
@ThomasWaldmann What about making a proper list out of this? Like a wiki page?

Here in this issue, it's not possible for one to invalidate an old offer and for future readers it's much easier to have a small comprehensive list instead of a thread where half of the stuff doesn't represent the current state.

**Comment by milkey-mouse** [comment]
> it's not possible for one to invalidate an old offer

I mean, technically somebody can edit/delete the original comment. I agree that's not a "scalable" strategy, though.

**Comment by enkore** [comment]
https://www.borgbackup.org/support/commercial.html

https://github.com/borgbackup/borgbackup.github.io/blob/source/support/commercial.rst

**Comment by ThomasWaldmann** [comment]
Important: do PRs against "source" there (as seen in 2nd url in previous comment), not "master".

**Comment by enkore** [comment]
You have to explicitly select "master" instead of "source" when creating PRs, btw.

**Comment by hi2u** [comment]
Is anyone using Borg on a time4vps storage VPS?

The latest version of Debian they offer is Debian 8 with kernel 2.6.32.x - and it's using OpenVZ, so you can't upgrade it yourself.

@ThomasWaldmann Any rough idea on how long you think new versions of Borg will continue to work under Debian 8 with kernel 2.6.32.x?

**Comment by jeanregisser** [comment]
@hi2u I'm using a 1 TB time4vps VPS. No issue sofar.

**Comment by ThomasWaldmann** [comment]
@hi2u borg is currently tested and built (standalone binary) on Debian 7. When LTS support for D7 ends, it is likely that we will switch to Debian 8 for that (and keep it there as long as D8 receives security updates / LTS support).

No idea about the old kernel.

**Comment by mannp** [comment]
Hetzner want 2.90 euro + vat for borgback storage box though, its not free?

borgbase has 100g free while in beta but seemingly no prices for outside of the beta; also based in Hong Kong by the look of it.

Anyone used borgbase?

**Comment by Vertux** [comment]
@mannp 
> Hetzner want 2.90 euro + vat for borgback storage box though, its not free?

As I already said, if you rent one of the "bigger" Hetzner Servers e.g. EX41-SSD you get 100GB Backup Storage with Borg Support for free. If you want just a Storage Box, you have to pay for it, they do not offer any storage for free. I personally have no experience with borgbase.

**Comment by mannp** [comment]
@Vertux Sorry to waste your time repeating yourself.

The fact that your links first task is to setup your 'storage box' is what threw me.

> Activate Borg and configure your storage box

**Comment by Vertux** [comment]
No need to apologize @mannp I should have made that clearer.

**Comment by noctuid** [comment]
Anyone here using borgbase now? I was considering using restic for online backups with backblaze, but borgbase's 2tb plan is even cheaper than backblaze b2 (12.5 a month vs 20 minimum for b2) and the same price per GB after that. I feel elation.

**Comment by hashbackup** [comment]
Backblaze B2 is $0.005/GB/mo, so 2TB is $10/mo.  There is no $20 minimum.

**Comment by noctuid** [comment]
My bad. I had upload/download in their calculator. If you don't have much upload/download backblaze is cheaper.

**Comment by hashbackup** [comment]
Uploading to B2 is free, and downloading is cheaper than any of the major players except Wasabi as far as I know.  But Wasabi has monthly minimums, minimum file sizes, delete penalties, and other price gimmicks that make their pricing all but impossible to predict or compare.  Downloads from B2 are $0.01/GB with a 1GB/day free download allowance.

Maybe you filled in the "monthly upload" box on their calculator.  In that cases, the charges go up because you are accumulating more data each month, ie, storage cost would increase every month.

**Comment by jcormier** [comment]
> backblaze is cheaper

borg still can't directly backup to backblaze correct?  So its not really an option.

**Comment by remyabel** [comment]
> borg still can't directly backup to backblaze correct? So its not really an option.

How so? Like any other storage provider where you don't have direct access you can just mount/rclone the files. Either create a new repo, then rclone it or rclone mount then create the repo inside the mount. You can then use the [cache](https://rclone.org/cache/) backend to make the mount more reliable.

**Comment by ThomasWaldmann** [comment]
@jcormier as far as borgbackup is concerned: correct, there is no cloud support.

Some people combine borg with rclone and let rclone deal with the cloud stuff.

**Comment by jcormier** [comment]
Thanks.  Is there some way to setup the rclone example to prevent a cryptolocker from encrypting the local backup files and then it getting automatically sync'd to the cloud?

I think with native cloud support in the backup software, if local files get encrypted they could still be reverted to a previous version since the database on the cloud wasn't encrypted

**Comment by milkey-mouse** [comment]
@jcormier Can't rclone [mount remotes with FUSE](https://rclone.org/commands/rclone_mount/)? Temporarily mounting the remote and unmounting it when the backup is done will give some protection (any cryptolocker not tailored to your specific system won't try to mount it—and if a custom virus is targeting you, you have bigger problems).

**Comment by willsALMANJ** [comment]
With some cloud providers, you can enable file versioning and create a read/write token that can not access file versioning (one option for S3: https://docs.aws.amazon.com/AmazonS3/latest/dev/Versioning.html#MultiFactorAuthenticationDelete), so a crypto-locker couldn't permanently destroy backup data.

**Comment by jcormier** [comment]
> With some cloud providers, you can enable file versioning

Wouldn't that take up a lot of extra cloud storage space though?  Does borg store its files in a way that could be snapshotted efficiently?

> Can't rclone [mount remotes with FUSE](https://rclone.org/commands/rclone_mount/)?

Hmm, I assume you'd need to set up caching for this to prevent lots of download charges.  And I suppose if the cache gets encrypted, you could delete it and let rclone recreate the cache.

**Comment by willsALMANJ** [comment]
> Wouldn't that take up a lot of extra cloud storage space though? Does borg store its files in a way that could be snapshotted efficiently?
It has been a while since I looked at the details of borg, so someone can correct me if I am wrong, but I believe borg stores all of the backup data and most of its metadata in append-only files that are written only once. So most of the files written to the cloud should only ever have one version. When you wanted to free up versioned space, you would have to go through and remove the saved versions of deleted repository files when you trusted that they hadn't been deleted maliciously.

I would be careful about download charges though. I don't know how good borg is about minimizing reads, since normal operation assumes borg has local access to the repository.

**Comment by milkey-mouse** [comment]
If you're syncing to a cloud service with its own file versioning system, what's the point of using borg?

**Comment by willsALMANJ** [comment]
Benefits of using borg with a cloud service include client side encryption, deduplication, compression, and more efficient versioning (because of deduplication).

**Comment by azzamsa** [comment]
IMHO, I think we need a place to share the experience of Borg user using these cloud services mentioned in [Paid Support and Services](https://www.borgbackup.org/support/commercial.html) or other services. So new user can decide faster and wiser.

Maybe a separate issue or any other recommendation?

Thank you.

**Comment by ThomasWaldmann** [comment]
#8405 - borg2 (beta / master branch) has just gained cloud storage support via rclone, 70+ new storage providers.

**Comment by ThomasWaldmann** [comment]
Considering that most stuff here is 6+ years old and we recommend the most popular options on the borg homepage, I am closing this.
